### PR TITLE
Enable adding and removing SSO config from web

### PR DESCRIPTION
**Pull request by fiskus**
# Description

Enable not only editing but also adding and removing the SSO config from Admin UI.

image

# TODO

- [x] [Changelog](../tree/master/docs/CHANGELOG.md) entry (skip if change is not significant to end users, e.g. docs only)

### CODE REVIEW

**Review by nl0 — DISMISSED** [review]
lgtm.
i'd also add an option to remove/reset the sso config (not sure if emptying the text field is a good enough ux), tho that seems out of scope

**Comment by fiskus** [comment]
Emptying the text field won't work actually, because the value is `"required"` now

**Comment by fiskus** [comment]
Deletion accompanied by https://github.com/quiltdata/quilt/pull/4098

File: catalog/app/containers/Admin/UsersAndRoles/SsoConfig.tsx [hunk 1]
```diff
@@ -79,19 +97,33 @@ function Form({
...
-            {error.message}
-          </Lab.Alert>
+        {submitFailed && (
+          <>
+            <FormError error={error || submitError} errors={FORM_ERRORS} />
+            <M.Typography variant="body2">
+              <StyledLink href={`${docs}/advanced/sso-permissions`} target="_blank">
+                See docs
+              </StyledLink>{' '}
+              to learn more on how to configure SSO mappings.
```
Comment (nl0) [inline catalog/app/containers/Admin/UsersAndRoles/SsoConfig.tsx:113]:
```suggestion
              Learn more about{' '}
              <StyledLink href={`${docs}/advanced/sso-permissions`} target="_blank">
                SSO permissions mapping
              </StyledLink>
              .
```

File: catalog/app/containers/Admin/UsersAndRoles/SsoConfig.tsx [hunk 2]
```diff
@@ -121,38 +153,70 @@ function Data({ children, close }: DataProps) {
...
-            return setError(new Error(`Unable to update SSO config: ${r.message}`))
+            return mkFormError(r.message)
           default:
-            assertNever(r)
+            return assertNever(r)
         }
       } catch (e) {
-        return setError(e instanceof Error ? e : new Error('Error updating SSO config'))
+        // eslint-disable-next-line no-console
+        console.error('Error updating policy')
```
Comment (nl0) [inline catalog/app/containers/Admin/UsersAndRoles/SsoConfig.tsx:181]: policy?

**Review by nl0 — DISMISSED** [review]

**Review by nl0 — APPROVED** [review]
lgtm, tho i feel like we need a confirmation dialog for the deletion

**Pull request merged**